src/Microsoft.AspNetCore.Hosting/WebHostBuilderExtensions.cs

```diff
@@ -92,5 +93,28 @@ public static IWebHostBuilder UseDefaultServiceProvider(this IWebHostBuilder hos
                 services.Replace(ServiceDescriptor.Singleton<IServiceProviderFactory<IServiceCollection>>(new DefaultServiceProviderFactory(options)));
             });
         }
+
+        /// <summary>
+        /// Configures and use a <see cref="LoggerFactory"/> for the web host.
+        /// </summary>
+        /// <param name="hostBuilder">The <see cref="IWebHostBuilder"/> to configure.</param>
+        /// <param name="configure">A callback used to configure the <see cref="LoggerFactory"/> that will be added as a singleton and used by the application.</param>
+        /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+        public static IWebHostBuilder ConfigureLogging(this IWebHostBuilder hostBuilder, Action<HostingContext, LoggerFactory> configure)
```

Comment:
Spoke offline, this is not the intended functionality since it overrides previous LoggerFactory while being called ConfigureLogging. Will update.